presto-main/src/main/java/com/facebook/presto/operator/SpillingUtils.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.PrestoException;
+
+import java.util.concurrent.Future;
+
+import static com.facebook.airlift.concurrent.MoreFutures.getFutureValue;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.lang.String.format;
+
+public class SpillingUtils
+{
+    private SpillingUtils() {}
+
+    public static void checkSpillSucceeded(Future spillInProgress)
+    {
+        try {
+            getFutureValue(spillInProgress);
+        }
+        catch (PrestoException prestoException) {
+            throw new PrestoException(prestoException::getErrorCode, prestoException.getMessage(), prestoException);
```

Comment:
nit : what's the difference compared with directly rethrowing it?

Reply:
The difference is that this way we get both the stack trace of the cause and the current stacktrace.  this is helpful for example when the error is EXCEEDED_SPILL_LIMIT, because it can show you where we exceeded the spill limit (which operator), and not just the stacktrace from the FileSingleStreamSpiller. In other cases the stacktrace from the cause could be more valuable, so we want both. Wrapping the exception is the only way I know of to do that.

Reply:
@rschlussel : But the exception itself is already a `PrestoException`? -- so doesn't it already have error code?